src/actions/posts.js

```diff
@@ -364,6 +364,26 @@ export function editPost(post) {
     });
 }
 
+export function getUnreadPostData(unreadChan, state) {
```

Comment:
Perhaps we can turn this into an action creator instead of just having a function that constructs the data portion of the action? It's dispatched with the same type in both places, so it might be cleaner to do it that way. That would also avoid passing around `state` since we'd just get it from thunk

Reply:
having a function that only generates the data portion makes it pure, as there is no side effects, easier to maintain and test if felt necessary.
having it dispatch the action, wouldn't prevent it to get the state, as the only thing this function does is to extract info from the state, so the only way to remove it is to have the caller function extract it, but then, all the redundancy saved was back in the function, so I'm not really sure what are we gaining with that change?
can we discuss this? since both of you see a benefit I don't I'd like to dig deeper to find what I'm probably missing (or maybe its just that I'm too used to functional programing 😆 )

Reply:
The main reason I was thinking it might be better to turn it into an action creator is that the caller wouldn't have to care as much about the arguments. For example, these two action creators are both called the same way, but one requires the state and the other doesn't:

```javascript
const myAction1 = (value) => ({type: 'MY_ACTION', value});
const myAction2 = (value) => (dispatch, getState) => {
    const otherValue = getState().otherValue;
    dispatch({type: 'MY_ACTION', value, otherValue);
};

dispatch(myAction1(123));
dispatch(myAction2(123));
```

I agree that it's kind of unnecessary to do that if you're wanting to keep it really pure though. In that case, we could make it so that instead of just returning the data portion of the action, it returns the entire object including the `type` field since it's the same in both places.

I'm just 0/5 on those either way though